backends/cuda-shared/ceed-cuda-shared-basis.c

```diff
@@ -0,0 +1,741 @@
+// Copyright (c) 2017-2018, Lawrence Livermore National Security, LLC.
+// Produced at the Lawrence Livermore National Laboratory. LLNL-CODE-734707.
+// All Rights reserved. See files LICENSE and NOTICE for details.
+//
+// This file is part of CEED, a collection of benchmarks, miniapps, software
+// libraries and APIs for efficient high-order finite element and spectral
+// element discretizations for exascale applications. For more information and
+// source code availability see http://github.com/ceed.
+//
+// The CEED research is supported by the Exascale Computing Project 17-SC-20-SC,
+// a collaborative effort of two U.S. Department of Energy organizations (Office
+// of Science and the National Nuclear Security Administration) responsible for
+// the planning and preparation of a capable exascale ecosystem, including
+// software, applications, hardware, advanced system engineering and early
+// testbed platforms, in support of the nation's exascale computing imperative.
+
+#include <ceed-impl.h>
+#include "../include/ceed.h"
+#include "ceed-cuda-shared.h"
+#include "../cuda/ceed-cuda.h"
```

Comment:
@jedbrown, I am doing this reference of another backend's header as well in the blocked backend. I am concerned about possible fragility. Thoughts?

Reply:
I don't think it's a priority, but we may eventually want to discuss how backends can expose a stable interface for extensions.